server/datastore/mysql/vpp.go

```diff
@@ -0,0 +1,103 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/jmoiron/sqlx"
+)
+
+func (ds *Datastore) BatchInsertVPPApps(ctx context.Context, apps []*fleet.VPPApp) error {
+	return ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		if err := insertVPPApps(ctx, tx, apps); err != nil {
+			return ctxerr.Wrap(ctx, err, "BatchInsertVPPApps insertVPPApps transaction")
+		}
+
+		return nil
+	})
+}
+
+func (ds *Datastore) InsertVPPAppWithTeam(ctx context.Context, app *fleet.VPPApp, teamID *uint) error {
+	return ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		if err := insertVPPApps(ctx, tx, []*fleet.VPPApp{app}); err != nil {
+			return ctxerr.Wrap(ctx, err, "InsertVPPAppWithTeam insertVPPApps transaction")
+		}
+
+		if err := insertVPPAppTeams(ctx, tx, app.AdamID, teamID); err != nil {
+			return ctxerr.Wrap(ctx, err, "InsertVPPAppWithTeam insertVPPAppTeams transaction")
+		}
+
+		return nil
+	})
+}
+
+func (ds *Datastore) GetAssignedVPPApps(ctx context.Context, teamID *uint) (map[string]struct{}, error) {
+	stmt := `
+SELECT 
+	adam_id
+FROM 
+	vpp_apps_teams vat
+WHERE
+	vat.global_or_team_id = ?
+	`
+	var tmID uint
+	if teamID != nil {
+		tmID = *teamID
+	}
+
+	var results []string
+	if err := sqlx.SelectContext(ctx, ds.reader(ctx), &results, stmt, tmID); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "get assigned VPP apps")
+	}
+
+	appSet := make(map[string]struct{})
+	for _, r := range results {
+		appSet[r] = struct{}{}
+	}
+
+	return appSet, nil
+}
+
+func insertVPPApps(ctx context.Context, tx sqlx.ExtContext, apps []*fleet.VPPApp) error {
+	stmt := `
+INSERT INTO vpp_apps
+	(adam_id, available_count, bundle_identifier, icon_url, name)
+VALUES
+%s
+ON DUPLICATE KEY UPDATE updated_at = CURRENT_TIMESTAMP
```

Comment:
> IIRC we need to store the version too? is that right?

yep, we do. @mna mentioned he was going to do it in his PR, but I'm not sure what the status of that is. Should we just add it here, especially since we wanna demo this on Friday if possible?

> I'm thinking to update more things on ON DUPLICATE KEY ? maybe icon_url, name and version if we add it too 

_definitely_ version, but all of those make sense!

Reply:
sounds good, yeah! If necessary I can pick up his PR once I finish mine and clean up any conflicts